README.md

```diff
@@ -1,135 +1,268 @@
+regular expressions, with options for replacement and capture. The
+extracted features are stored in the database and can contribute to the
+valid/invalid decisions via the decision DSL mentioned in the previous
+section.
+
+TODO: Example ?
+
+### File-Detail Plugins
+
+File detail view plugins in the FAW visualize file contents by rendering
+them as HTML or other browser-displayable formats. Similar to parser
+plugins, they execute an external command that transforms the file
+content and streams it to the standard output. The FAW captures and
+renders this content in the browser based on the provided MIME type. An
+example of a file detail view plugin is PolyFile, which creates
+interactive HTML reports for various file formats, enabling users to
+view and search data structures within the file.
+
+### Corpus-Level Plugins
+
+Decision plugins in the FAW operate on the entire corpus, aggregating
+and visualizing information about dialects or common patterns. Like
+other plugins, they execute external commands, but can access the
+complete corpus and features captured during parser invocations.
+
+Results from the command are expected to be streamed in JSON line
+format, and the FAW can be optionally configured to capture and render
+custom HTML files generated by the plugin. This flexibility allows
+decision plugins to introduce new UI elements and actions within the
+FAW. FAW's root cause analysis are implemented as decision plugins.
+
+### Pipeline Plugins
+
+The FAW supports complex plugins with multiple stages and the capability
+to generate new data-driven parsers based on corpus inputs.
+
+TODO: Point to plugin tutorial
+
+### Transforming Inputs and Universal Output Parsers
+
+The FAW allows for input transformers that dynamically modify inputs
+before processing them. These transformers take a file from the corpus
+as input run external commands that produce transformed content on the
+standard output. Transformed inputs are treated as transient and not
+permanently stored. The SDF distribution, for example, use input
+transformers to generate a large corpus of SDF files from very simple
+seed input.
+
+The FAW also supports universal parser parsers, which operate on the
+output of all configured parsers, generating additional features. These
+parsers run external commands that accept specially formatted streams
+and emit JSON dictionaries of new features. Universal parser parsers
+unify format-specific considerations across all parsers
+
+## Root Cause Analysis
+
+TODO: Ask!
+
+## Rendering differentials
+
+The FAW includes a rendering differential detection algorithm designed
+for file formats that generate on-screen images like PDFs. Unlike
+existing algorithms that often report false positives, this algorithm
+reduces such errors and accurately identifies true differentials,
+enabling the identification of root causes for the differences.
+
+## PolyFile / PolyTracker Integration
```

Comment:
For this section, is it worth acknowledging Trail of Bits explicitly for their contributions and adding links to the referenced PolyFile and PolyTracker repositories?

Reply:
Links have been added. as of commit [c7d9153](https://github.com/GaloisInc/FAW/pull/94/commits/c7d9153ac4aacdbffc60c38bb19d27d9908c8ad9).